sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -593,7 +592,6 @@ object DataSource extends Logging {
       "org.apache.spark.ml.source.libsvm.DefaultSource" -> libsvm,
       "org.apache.spark.ml.source.libsvm" -> libsvm,
       "com.databricks.spark.csv" -> csv,
-      "com.databricks.spark.avro" -> avro,
```

Comment:
@gengliangwang, not a big deal but how about adding the entry at 618 here conditionally since this is called backward compatibility map?

Reply:
@HyukjinKwon I did add it in the `backwardCompatibilityMap` at first. But later on I find that the configuration won't be effective in run time, since the `backwardCompatibilityMap` is a `val`.  (We can change `backwardCompatibilityMap` to method to resolve that.) Also the code looks ugly.
```
val ret = Map(...)
if(...) {
 ret + k -> v
} else {
 ret
}
// it would be worse if we have more configurations.
```

Reply:
Ah okie makes sense if there's a reason.